dbt_subprojects/dex/macros/models/_project/uniswap_compatible_liquidity.sql

```diff
@@ -235,237 +235,226 @@ get_prices as (
-        group by 1, 2, 3, 4, 7
+    calls_decoded as (
+        SELECT
+            c.*,
+
+            -- ---- decode callerDelta (top/bottom 16 bytes, sign-extended to int256) ----
+            CASE
+                WHEN BITWISE_AND(VARBINARY_TO_BIGINT(VARBINARY_SUBSTRING(c.callerDelta_vb, 1, 1)), FROM_BASE('80', 16)) = FROM_BASE('80', 16)
+                THEN VARBINARY_TO_INT256(VARBINARY_CONCAT(FROM_HEX('0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF'), VARBINARY_SUBSTRING(c.callerDelta_vb, 1, 16)))
+                ELSE VARBINARY_TO_INT256(VARBINARY_CONCAT(FROM_HEX('0x00000000000000000000000000000000'), VARBINARY_SUBSTRING(c.callerDelta_vb, 1, 16)))
+            END AS callerDelta_token0,
+
+            CASE
+                WHEN BITWISE_AND(VARBINARY_TO_BIGINT(VARBINARY_SUBSTRING(c.callerDelta_vb, 17, 1)), FROM_BASE('80', 16)) = FROM_BASE('80', 16)
+                THEN VARBINARY_TO_INT256(VARBINARY_CONCAT(FROM_HEX('0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF'), VARBINARY_SUBSTRING(c.callerDelta_vb, 17, 16)))
+                ELSE VARBINARY_TO_INT256(VARBINARY_CONCAT(FROM_HEX('0x00000000000000000000000000000000'), VARBINARY_SUBSTRING(c.callerDelta_vb, 17, 16)))
+            END AS callerDelta_token1,
+
+            -- ---- decode feesAccrued (top/bottom 16 bytes, sign-extended to int256) ----
+            CASE
+                WHEN BITWISE_AND(VARBINARY_TO_BIGINT(VARBINARY_SUBSTRING(c.feesAccrued_vb, 1, 1)), FROM_BASE('80', 16)) = FROM_BASE('80', 16)
+                THEN VARBINARY_TO_INT256(VARBINARY_CONCAT(FROM_HEX('0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF'), VARBINARY_SUBSTRING(c.feesAccrued_vb, 1, 16)))
+                ELSE VARBINARY_TO_INT256(VARBINARY_CONCAT(FROM_HEX('0x00000000000000000000000000000000'), VARBINARY_SUBSTRING(c.feesAccrued_vb, 1, 16)))
+            END AS feesAccrued_token0,
+
+            CASE
+                WHEN BITWISE_AND(VARBINARY_TO_BIGINT(VARBINARY_SUBSTRING(c.feesAccrued_vb, 17, 1)), FROM_BASE('80', 16)) = FROM_BASE('80', 16)
+                THEN VARBINARY_TO_INT256(VARBINARY_CONCAT(FROM_HEX('0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF'), VARBINARY_SUBSTRING(c.feesAccrued_vb, 17, 16)))
+                ELSE VARBINARY_TO_INT256(VARBINARY_CONCAT(FROM_HEX('0x00000000000000000000000000000000'), VARBINARY_SUBSTRING(c.feesAccrued_vb, 17, 16)))
+            END AS feesAccrued_token1
+        FROM 
+        get_calls c
     ),
 
-    modify_events as (
-        select 
-            evt_block_time as block_time
-            , evt_block_number as block_number 
-            , id 
-            , evt_tx_hash as tx_hash 
-            , evt_index 
-            , amount0 
-            , amount1 
-        from 
-        modify_liquidity_events
-        where liquidityDelta <= int256 '0'
-    ),
 
-    single_pools as (
-        select 
-            tx_hash 
-            , count(distinct id) as num_pools 
+    evts as (
+        select
+            contract_address,
+            evt_tx_hash,
+            evt_tx_from,
+            evt_block_time,
```

Comment:
```suggestion
            evt_block_time,
            evt_block_date,
```

Reply:
you could also carry this field forward into final select statements (i.e. you use `block_time` to build `block_date` in final select), but it's more critical to use in the join conditions below.

Reply:
got it, thank you ser!